src/services/saneKeyboardEvents.util.js

```diff
@@ -140,20 +149,25 @@ var saneKeyboardEvents = (function() {
     }
 
     function handleKey() {
-      handlers.keystroke(stringify(keydown), keydown);
+      if (isandroid && keydown.which === 229) {
+        androidcompose = true;
+        textarea.val('\0');
+      } else {
+        handlers.keystroke(stringify(keydown), keydown);
+      }
     }
 
     // -*- event handlers -*- //
     function onKeydown(e) {
       keydown = e;
       keypress = null;
 
-      if (shouldBeSelected) checkTextareaFor(function(e) {
+      if (shouldBeSelected) checkTextareaOnce(function(e) {
         if (!(e && e.type === 'focusout') && textarea[0].select) {
-          textarea[0].select(); // re-select textarea in case it's an unrecognized
+          // re-select textarea in case it's an unrecognized key that clears
+          // the selection, then never again, 'cos next thing might be blur
+          textarea[0].select();
         }
-        checkTextarea = noop; // key that clears the selection, then never
-        clearTimeout(timeoutId); // again, 'cos next thing might be blur
       });
```

Comment:
```js
if (shouldBeSelected) {
    checkTextareaOnce(function(e) {
        if (!(e && e.type === 'focusout') && textarea[0].select) {
          // re-select textarea in case it's an unrecognized key that clears
          // the selection, then never again, 'cos next thing might be blur
          textarea[0].select();
        }
    });
}
```

Reply:
Ah, gotcha. Well, the code in question was written by the creator of Mathquill and consistent with internal style-- I just copied the changes.

I expect in the future we will abandon this code and work with a better-maintained fork. The only benefit our fork has is we committed build artifacts and manually made the output a CommonJS module.